### PR TITLE
Add extensions to several asset operations

### PR DESCRIPTION
**Pull request by abitmore**
For BSIP 48, 75 and 77

Features:
* price feed producers can publish ICR (supplement to BSIP77)
* asset owners can update MCR and MSSR directly (BSIP75) (also ICR due to BSIP77)
* asset precision can be updated when asset has zero supply (BSIP48)
  * precision of prediction markets can't be updated
  * if any other asset is backed by this asset, this asset's precision can't be updated
* when updating asset, can skip updating CER (core_exchange_rate), this is useful for updating via proposals which take time to approve (may be put into BSIP48)
* new flags for assets (BSIP48)
  * whether max supply can be updated
  * whether new supply can be created (issued by asset owner for UIAs, or created by borrowers for MPAs/PMs)
* new asset issuer permissions (to be forfeited by asset owners) (BSIP48/75/77)
  * to update the new "whether max supply can be updated" flag
  * to update the new "whether new supply can be created" flag
  * for MPAs/PMs, to update MCR
  * for MPAs/PMs, to update ICR
  * for MPAs/PMs, to update MSSR
* fix a bug about `flags`
  * before the hard fork, unknown bits can be set (to `1`) in `flags` when creating an asset, and can not be unset (to `0`) when updating the asset (and the non-buggy part: can not set the unknown bits when updating);
  * after the hard fork, when creating an asset, neither the invalid bits nor the unknown bits can be set in `flags`;
  * after the hard fork, when updating an asset, neither the invalid bits nor the unknown bits can be set in `flags`; if a bit was already set, when updating the asset, it must be unset.
* fix a bug about `max_supply`
  * before the hard fork, can update an asset's max supply to a number smaller than current supply
  * after the hard fork, can't update an asset's max supply to a number smaller than current supply
* fix a bug about `global_settle` permission of prediction markets
  * before the hard fork, the `global_settle` permission of prediction markets can be disabled via `asset_update_operation`, if done so, it can't be enabled again if supply is not zero;
  * after the hard fork, when updating a prediction market, the `global_settle` permission must be enabled in `new_options`, and it's allowed to be enabled if was disabled before the hard fork.

TODO:
- [x] test case for new precision
- [x] test case for ICR/MCR/MSSR

### CODE REVIEW

File: libraries/protocol/include/graphene/protocol/asset_ops.hpp [hunk 1]
```diff
@@ -392,13 +409,19 @@ namespace graphene { namespace protocol {
     */
    struct asset_publish_feed_operation : public base_operation
    {
+      struct ext
+      {
+         /// After BSIP77, price feed producers can feed ICR too
```
Comment (MichelSantos) [inline libraries/protocol/include/graphene/protocol/asset_ops.hpp:414]: Please note that the current spec for BSIP77 describes that asset owners can update the ICR.  But it makes no mention about feed producers.

Comment (abitmore) [reply libraries/protocol/include/graphene/protocol/asset_ops.hpp:414]: Yes I'm aware that it's not in the specification. But I think it makes sense so I implemented it anyway.

Comment (MichelSantos) [reply libraries/protocol/include/graphene/protocol/asset_ops.hpp:418]: Fair enough.  It does it make it consistent with how MCR and MSSR are handled.

**Comment by abitmore** [comment]
Ready for review.

**Pull request merged**